File: Palindrome Number.py
class Solution:
    # @return a boolean
    def isPalindrome(self, x):
        string=str(x)
        i=0
        flag=True
        while i<len(string)/2:
            if string[i]!=string[-i-1]:
                flag=False
                return False
            else:
                i+=1
        return flag
s=Solution()
print s.isPalindrome(1221)

File: PalindromeNumber.java
public class PalindromeNumber {
    public boolean isPalindrome(int x) {
        if(x>=Integer.MAX_VALUE||x<=Integer.MIN_VALUE||x<0)
            return false;
        int cur=x;
        int count=0;
        while(x>0){
            count=10*count+(x%10);
            x/=10;
        }
        return cur==count;
    }
}